jadx-core/src/main/java/jadx/core/deobf/Deobfuscator.java

```diff
@@ -107,6 +132,237 @@ public void savePresets() {
+		MemoryMappingTree mappingTree = new MemoryMappingTree();
+		// Map < SrcName >
+		Set<String> mappedClasses = new HashSet<>();
+		// Map < DeclClass + ShortId >
+		Set<String> mappedFields = new HashSet<>();
+		Set<String> mappedMethods = new HashSet<>();
+		Set<String> methodsWithMappedElements = new HashSet<>();
+		// Map < DeclClass + MethodShortId + CodeRef, NewName >
+		Map<String, String> mappedMethodArgsAndVars = new HashMap<>();
+		// Map < DeclClass + *ShortId + *CodeRef, Comment >
+		Map<String, String> comments = new HashMap<>();
+
+		// We have to do this so we know for sure which elements are *manually* renamed
+		for (ICodeRename codeRename : codeData.getRenames()) {
+			if (codeRename.getNodeRef().getType().equals(RefType.CLASS)) {
+				mappedClasses.add(codeRename.getNodeRef().getDeclaringClass());
+			} else if (codeRename.getNodeRef().getType().equals(RefType.FIELD)) {
+				mappedFields.add(codeRename.getNodeRef().getDeclaringClass() + codeRename.getNodeRef().getShortId());
+			} else if (codeRename.getNodeRef().getType().equals(RefType.METHOD)) {
+				if (codeRename.getCodeRef() == null) {
+					mappedMethods.add(codeRename.getNodeRef().getDeclaringClass() + codeRename.getNodeRef().getShortId());
+				} else {
+					methodsWithMappedElements.add(codeRename.getNodeRef().getDeclaringClass() + codeRename.getNodeRef().getShortId());
+					mappedMethodArgsAndVars.put(codeRename.getNodeRef().getDeclaringClass()
+							+ codeRename.getNodeRef().getShortId()
+							+ codeRename.getCodeRef(),
+							codeRename.getNewName());
+				}
+			}
+		}
+		for (ICodeComment codeComment : codeData.getComments()) {
+			comments.put(codeComment.getNodeRef().getDeclaringClass()
+					+ (codeComment.getNodeRef().getShortId() == null ? "" : codeComment.getNodeRef().getShortId())
+					+ (codeComment.getCodeRef() == null ? "" : codeComment.getCodeRef()),
+					codeComment.getComment());
+			if (codeComment.getCodeRef() != null) {
+				methodsWithMappedElements.add(codeComment.getNodeRef().getDeclaringClass() + codeComment.getNodeRef().getShortId());
+			}
+		}
+
+		try {
+			if (mappingFormat.hasSingleFile()) {
+				if (path.toFile().exists()) {
+					path.toFile().delete();
+				}
+				path.toFile().createNewFile();
+			} else {
+				FileUtils.makeDirs(path);
+			}
+
+			mappingTree.visitHeader();
+			mappingTree.visitNamespaces("official", Arrays.asList("named"));
+			mappingTree.visitContent();
+
+			for (ClassNode cls : root.getClasses()) {
+				ClassInfo classInfo = cls.getClassInfo();
+				String classPath = classInfo.makeRawFullName().replace('.', '/');
+				String rawClassName = classInfo.getRawName();
+
+				if (classInfo.hasAlias()
+						&& !classInfo.getAliasShortName().equals(classInfo.getShortName())
+						&& mappedClasses.contains(rawClassName)) {
+					mappingTree.visitClass(classPath);
+					String alias = classInfo.makeAliasRawFullName().replace('.', '/');
+
+					if (alias.length() >= Consts.DEFAULT_PACKAGE_NAME.length()
```

Comment:
Idea suggest to use `alias.startsWith(Consts.DEFAULT_PACKAGE_NAME)` instead this :)

Reply:
That would be easier, indeed xD

Reply:
Resolved by 782a56c482a0a382b1d0f2c9c6bf4b38f4593145